src/test/java/com/puppycrawl/tools/checkstyle/checks/metrics/NPathComplexityCheckTest.java

```diff
@@ -146,4 +175,51 @@ public void testDefaultHooks() {
         npathComplexityCheckObj.visitToken(ast);
         npathComplexityCheckObj.leaveToken(ast);
     }
+
+    @Test
+    public void testVisitTokenBeforeExpressionRange() {
+        // Create first ast
+        final DetailAST astIf = mockAST(TokenTypes.LITERAL_IF, "if", "mockfile", 2, 2);
+        final DetailAST astIfLeftParen = mockAST(TokenTypes.LPAREN, "(", "mockfile", 3, 3);
+        astIf.addChild(astIfLeftParen);
+        final DetailAST astIfTrue =
+                mockAST(TokenTypes.LITERAL_TRUE, "true", "mockfile", 3, 3);
+        astIf.addChild(astIfTrue);
+        final DetailAST astIfRightParen = mockAST(TokenTypes.RPAREN, ")", "mockfile", 4, 4);
+        astIf.addChild(astIfRightParen);
+        // Create ternary ast
+        final DetailAST astTernary = mockAST(TokenTypes.QUESTION, "?", "mockfile", 1, 1);
+        final DetailAST astTernaryTrue =
+                mockAST(TokenTypes.LITERAL_TRUE, "true", "mockfile", 1, 2);
+        astTernary.addChild(astTernaryTrue);
+
+        final NPathComplexityCheck mock = new NPathComplexityCheck();
+        // visiting first ast, set expressionSpatialRange to [2,2 - 4,4]
+        mock.visitToken(astIf);
+        //visiting ternary, it lies before expressionSpatialRange
+        mock.visitToken(astTernary);
```

Comment:
We don't assert anything in this method nor throw any exceptions, which will lead to false security.
Can we assert something to prove what this test does and make sure we don't break it in the future?